src/webview/notifications.ts

```diff
@@ -31,52 +31,29 @@ export class NotificationsHandler {
 }
 
 export const notificationsClassDefinition = `(() => {
-  class Notification {
-    static permission = 'granted';
-
-    constructor(title = '', options = {}) {
-      this.title = title;
-      this.options = options;
-      try {
-        window.ferdium.displayNotification(title, options)
-          .then(() => {
-            if (typeof (this.onClick) === 'function') {
-              this.onClick();
-            }
-          });
-      } catch(error) {
-	        this.options.onClick = null;
-          window.ferdium.displayNotification(title, options)
-            .then(() => {
-              if (typeof (this.onClick) === 'function') {
-                this.onClick();
-              }
-            });
-      }
-    }
-
-    static requestPermission(cb = null) {
-      if (!cb) {
-        return new Promise((resolve) => {
-          resolve(Notification.permission);
-        });
-      }
-
-      if (typeof (cb) === 'function') {
-        return cb(Notification.permission);
-      }
-
-      return Notification.permission;
-    }
+class Notification extends window.Notification {
+  static permission = 'granted';
+
+  constructor(title = '', options = {}) {
+    super(title, options);
+    window.ferdium.displayNotification(title, options).then(() => {
+      // Continue to dispatch the event to the original Notification class
+      super.dispatchEvent(new Event('click'));
+    });
+  }
 
-    onNotify(data) {
-      return data;
+  static requestPermission(cb) {
+    if (typeof cb === 'function') {
+      cb(Notification.permission);
```

Comment:
does this need to `return` the returned value from the `cb` call?

Reply:
that's a great question, I'm not entirely sure if yes or no... Previously we were returning, maybe we can see if the same still works

Reply:
I think the callback is deprecated (but leaving here for old website implementations)
https://developer.mozilla.org/en-US/docs/Web/API/Notification/requestPermission_static